dipy/workflows/align.py

```diff
@@ -45,16 +61,576 @@ def run(self, input_files, new_vox_size, order=1, mode='constant', cval=0,
-            
             data, affine, vox_sz = load_nifti(inputfile, return_voxsize=True)
             logging.info('Processing {0}'.format(inputfile))
             new_data, new_affine = reslice(data, affine, vox_sz, new_vox_size,
                                            order, mode=mode, cval=cval,
                                            num_processes=num_processes)
             save_nifti(outpfile, new_data, new_affine)
             logging.info('Resliced file save in {0}'.format(outpfile))
-        
\ No newline at end of file
+
+
+class ImageRegistrationFlow(Workflow):
+    """
+    The registration workflow is organized as a collection of different
+    functions. The user can intend to use only one type of registration
+    (such as center of mass or rigid body registration only).
+
+    Alternatively, a registration can be done in a progressive manner.
+    For example, using affine registration with progressive set to 'True'
+    will involve center of mass, translation, rigid body and full affine
+    registration. Whereas, when progressive is False the registration will
+    include only center of mass and affine registration. The progressive
+    registration will be slower but will improve the quality of the results.
+
+    This can be controlled by using the progressive flag (True by default).
+    """
+
+    def process_image_data(self, static_img, moved_img):
+        """
+        Function for preprocessing the image data. It involves
+        normalizing and copying static, moving images in the
+        red and green channel.
+
+        return: the normalized image data and range of pixels to be
+        selected.
+        """
+
+        static_img = 255 * ((static_img - static_img.min()) /
+                            (static_img.max() - static_img.min()))
+        moved_img = 255 * ((moved_img - moved_img.min()) /
+                           (moved_img.max() - moved_img.min()))
+
+        # Create the color images
+        overlay = np.zeros(shape=(static_img.shape) + (3,), dtype=np.uint8)
+        overlay[..., 0] = static_img
+        overlay[..., 1] = moved_img
+        mean, std = overlay[overlay > 0].mean(), overlay[overlay > 0].std()
+        value_range = (mean - 0.5*std, mean + 0.05*std)
+
+        return overlay, value_range
+
+    def get_row_cols(self, num_slices):
+
+        """
+        Experimetal helper function to get the number
+        of rows and columns for the mosaic.
+
+        num_slices: int
+            The number of slices as obtained from the
+            create_mosaic function.
+        return: the number of rows and columns.
+        """
+        rows, cols = 0, 0
+
+        while True:
+            if num_slices % 5 == 0:
+                break
+            num_slices += 1
+
+        for i in range(5, num_slices):
+
+            if num_slices % i == 0:
+                rows = i
+                cols = num_slices // i
+                break
+
+        return rows, cols
```

Comment:
Hey, I already commented on this function in another file! We rarely have to copy code. DRY. In fact, I recognize some of the code in this file. What's happening? Can't you use common functions?

Reply:
So here is the thing: This branch was originally part of the affine_registration PR 1581 and so you will remember that you suggested a good number of improvements to that PR. I did all those changes to that PR and I was hoping that it will be merged. Meanwhile, I continued working on this branch thinking that when PR 1581 will be merged, I can just rebase and use all the latest changes here. 

But, this branch reached its's completion and the PR 1581 is still not merged so a large number of changes that you said, for example, using assert and not npt.assert_equal or not using () while importing or sticking to just one standard for documentation in the functions have already been done in PR 1581. nevertheless, i manually copied all the changes from my local repo for PR 1581 and put them here so that they align with all the improvements that I made in PR 1581. 

Sorry for the inconvenience and thanks very much for the useful comments :).